### PR TITLE
[T7][W09-B4]

### PR DESCRIPTION
**Pull request by dylansyj**
Created a UserGuide V0.0 for doTASK project

### CODE REVIEW

File: docs/UserGuide (original).md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# AddressBook Level 4 - User Guide
```
Comment (hwkchia) [inline docs/UserGuide (original).md]: If for any reason you want to keep the original guide for reference, may I suggest that you don't stage it?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -1,186 +1,239 @@
...
-   * **`exit`** : exits the app
-6. Refer to the [Features](#features) section below for details of each command.<br>
+3. Double-click on the file to start the application. The GUI should appear in a few seconds.<br>
+> <img src="images/startup_screen.jpg" width="600">
 
+4. Get started by adding your first task that you have in mind! Refer to the "2. Features" for further instructions. <br>
+> <img src="images/add_screen.jpg" width="600">
+
+
+5. After step 6, you can view your task in 3 different tabs.
```
Comment (hwkchia) [inline docs/UserGuide.md]: If you look at the rendered version, this IS step 6.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -1,186 +1,239 @@
...
+
+   > **Today/Date**
+   >
+   > * Shows you the list of tasks you have for the specified date. By default, it will show the tasks for today. Click left/right arrow to navigate between the dates.
```
Comment (hwkchia) [inline docs/UserGuide.md]: What do you mean by "click left/right arrow"?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -1,186 +1,239 @@
...
+   > <img src="images/today_screen.jpg" width="600">
```
Comment (hwkchia) [inline docs/UserGuide.md]: And where's the newly added CS3230 assignment on the mockup?  It also does not appear in the "Priority" and "Overview" tabs?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -1,186 +1,239 @@
...
+   >
+   > **Overview**
+   >
+   > * Shows you an overall view of the dates. The highest priority tasks will be shown as a preview for every single date. Scroll to navigate between the dates.
```
Comment (hwkchia) [inline docs/UserGuide.md]: Scroll?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -1,186 +1,239 @@
...
+   > <img src="images/overview_screen.jpg" width="600">
+   >
+   > **Priority**
+   >
+   > * Shows you 4 options of priority tabs, from highest to lowest, which are indicated by different colours. Selecting the tab allows you to view the list of tasks that is under that priority. Eg. By clicking on â€œHighest Priorityâ€� tab, it will bring you to the list of tasks that you have in it.<br>
```
Comment (hwkchia) [inline docs/UserGuide.md]: Something does not look right here... and which colours are for which priority?

File: docs/UserGuide.md [hunk 7]
```diff
@@ -1,186 +1,239 @@
...
-Format: `help`
-
-> Help is also shown if you enter an incorrect command e.g. `abcd`
+> Help is shown if you enter an incorrect command e.g. `asdf` <br>
+> Alternatively, type /help to obtain a list of commands that you can use.
 
-### 2.2. Adding a person: `add`
+### 2.2. Adding a Task : `add`
+Adds a task to the task manager. <br>
+Format: `add TASK_NAME d/DEADLINE p/PRIORITY_LEVEL [i/ANY_INFO] [t/TAGS]...`
```
Comment (hwkchia) [inline docs/UserGuide.md]: Refrain from using command line switches or flags.  Although it is easier to parse this way, part of the project requirements is to make the command flexible and as natural as possible.  We expect a fair amount of work in the parser, which is what makes it different from addressbook4.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -1,186 +1,239 @@
...
+   > `DEADLINE` will be in the format of "date month (indicated by first three letters) year"
```
Comment (hwkchia) [inline docs/UserGuide.md]: Make date and time formats flexible also.

File: docs/UserGuide.md [hunk 9]
```diff
@@ -1,186 +1,239 @@
...
+* add CS3230 Assignment 1 d/8 Jan 2018 p/1 i/How to do? t/School t/CS3230
+* add Buy Milk d/23 Feb 2017 p/4 t/Chores
 
-> Persons can have any number of tags (including 0)
+### 2.3. Listing the tasks: `list`
 
-Examples:
+Shows a list of all the tasks in the task manager.<br>
+> Listing is done automatically when user switches between the tabs.<br>
+> However, user can specify to list all the tasks he has in, which will be shown in a pop up.<br>
```
Comment (hwkchia) [inline docs/UserGuide.md]: what do you mean by "all the task he has in"?

File: docs/UserGuide.md [hunk 10]
```diff
@@ -1,186 +1,239 @@
...
+> User can also sort it accordingly to the deadlines, priorities, etc.
```
Comment (hwkchia) [inline docs/UserGuide.md]: Explain the sorting feature later, if there is one...

File: docs/UserGuide.md [hunk 11]
```diff
@@ -1,186 +1,239 @@
...
+   > `DEADLINE` will be in the format of "date month (indicated by first three letters) year"
+   > `PRIORITY_LEVEL` should be defined by integers 1 (high) to 4(low).<br>
+   > `ANY_INFO` allows you to key in details with regards to the task, optional.<br>
+   > `TAGS` allows you to assign tags to the tasks, optional.
 
-Adds a person to the address book<br>
-Format: `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...`
+Examples:
+* add CS3230 Assignment 1 d/8 Jan 2018 p/1 i/How to do? t/School t/CS3230
+* add Buy Milk d/23 Feb 2017 p/4 t/Chores
```
Comment (hwkchia) [inline docs/UserGuide.md]: How about floating tasks and timed tasks (with a from and to)?  You have to explicitly support these three types of tasks.

File: docs/UserGuide.md [hunk 12]
```diff
@@ -1,186 +1,239 @@
...
+Shows a list of the tasks sorted by a stated priority level, from 1 - 4.<br>
+Format: `list priority PRIORITY_LEVEL`
+> Tasks can be given any priority level from 0 to 3.
+> Tasks with PRORITY_LEVEL priority will be displayed.
 
-Edits an existing person in the address book.<br>
-Format: `edit INDEX [NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]...`
+Examples:
+* `list priority 1`
+* `list priority 2`
```
Comment (hwkchia) [inline docs/UserGuide.md]: These two examples look the same?

File: docs/UserGuide.md [hunk 13]
```diff
@@ -1,186 +1,239 @@
...
+> The tasks listed will be in clusters according to tags, but sorted in alphabetical order.<br>
 
 Examples:
+* `list t/CS3230 t/Work`
 
-* `edit 1 p/91234567 e/johndoe@yahoo.com`<br>
-  Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@yahoo.com` respectively.
+### 2.4. Editing an existing task: `edit`
+Edits an existing task in the task manager.<br>
+Format: `edit i/INDEX [n/TASK_NAME] [d/DEADLINE] [p/PRIORITY_LEVEL] [i/ANY_INFO] [t/TAGS]...`
```
Comment (hwkchia) [inline docs/UserGuide.md]: Where does one obtain the `INDEX` associated with a task (likewise for `delete` and `complete`?  From `list` command?

File: docs/UserGuide.md [hunk 14]
```diff
@@ -1,186 +1,239 @@
...
+> * Edits the task labeled by the INDEX digit as shown on the screen. The INDEX must be a positive integer, e.g. 1, 2, 3, ...
+> * At least one of the optional [fields] must be provided.
+> * Existing fields will be overwritten.
 
-### 2.5. Finding all persons containing any keyword in their name: `find`
+Examples:
+* `edit i/1 n/Assignment 2 d/25 Feb 2018 p/2`
+* `edit i/4 n/Exercise`
+
+### 2.5. Shifting tabs: `switch`
```
Comment (hwkchia) [inline docs/UserGuide.md]: Explaining `switch` here seems rather abrupt.  May I suggest to moving it to the front, even before `add`?

File: docs/UserGuide.md [hunk 15]
```diff
@@ -1,186 +1,239 @@
...
+> A general overview of performance/efficiency based on the usage of the task manager will be shown.<br>
 
-* `list`<br>
-  `delete 2`<br>
-  Deletes the 2nd person in the address book.
-* `find Betsy`<br>
-  `delete 1`<br>
-  Deletes the 1st person in the results of the `find` command.
+Format: `progress NUMBER_OF_DAYS`
+> NUMBER_OF_DAYS is a interger which indicates the your progress for these past days (excluding today).<br>
```
Comment (hwkchia) [inline docs/UserGuide.md]: Check for typo and grammar?

File: docs/UserGuide.md [hunk 16]
```diff
@@ -1,186 +1,239 @@
...
+* `complete 2`
+* `complete 5`
 
-> Deletes the person at the specified `INDEX`. <br>
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+### 2.8. Track overall progress: `progress`
 
-Examples:
+Shows all the tasks completed over time and also specifying the tasks overdued.<br>
```
Comment (hwkchia) [inline docs/UserGuide.md]: overdue.

File: docs/UserGuide.md [hunk 17]
```diff
@@ -1,186 +1,239 @@
...
-* `find Betsy`<br>
-  `delete 1`<br>
-  Deletes the 1st person in the results of the `find` command.
+Format: `progress NUMBER_OF_DAYS`
+> NUMBER_OF_DAYS is a interger which indicates the your progress for these past days (excluding today).<br>
+> It will provide an analysis on the number of task that you have completed/overdue.
 
-### 2.7. Select a person : `select`
+Examples:
+* `progress 7`
```
Comment (hwkchia) [inline docs/UserGuide.md]: I don't really understand the progress "feature"... maybe a nice-to-have?

File: docs/UserGuide.md [hunk 18]
```diff
@@ -1,186 +1,239 @@
...
 
-> Selects the person and loads the Google search page the person at the specified `INDEX`.<br>
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+Searches for tasks based on keywords by user's input.<br>
 
-Examples:
+Format: `search KEYWORD`
+> A valid input needs to be keyed in. No special characters such as ASCII whitespace is allowed.<br>
+> A popout will appear to show you the list of tasks that contains the stated `KEYWORD`
```
Comment (hwkchia) [inline docs/UserGuide.md]: You seem to like to use "popouts".  How do you close them?  Consider how Jim would like to deal with them.

File: docs/UserGuide.md [hunk 19]
```diff
@@ -1,186 +1,239 @@
...
-Format: `select INDEX`
+### 2.9. Search for tasks: `search`
 
-> Selects the person and loads the Google search page the person at the specified `INDEX`.<br>
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+Searches for tasks based on keywords by user's input.<br>
 
-Examples:
+Format: `search KEYWORD`
```
Comment (hwkchia) [inline docs/UserGuide.md]: Searching only on one keyword?  How about case-insensitivity?  How about partial search?

File: docs/UserGuide.md [hunk 20]
```diff
@@ -1,186 +1,239 @@
...
+* `search potato`
 
-### 2.8. Clearing all entries : `clear`
+### 2.10. Undo previous action : `undo`
 
-Clears all entries from the address book.<br>
-Format: `clear`
+Undos previous action that user made.<br>
+Format : `undo`
+> Any previous valid input will be reversed.
```
Comment (hwkchia) [inline docs/UserGuide.md]: Define "valid input"?

File: docs/UserGuide.md [hunk 21]
```diff
@@ -1,186 +1,239 @@
...
+Format : `redo`
+> Any previous `undo` will be reversed.
 
-### 2.10. Saving the data
+### 2.12. Saving the tasks
 
-Address book data are saved in the hard disk automatically after any command that changes the data.<br>
-There is no need to save manually.
+Upon creation of tasks, the tasks will be automatically saved in the folder where the program is held in.
+> Do not erase the saved data as it will result in a complete loss of data that cannot be recovered by the application itself.
```
Comment (hwkchia) [inline docs/UserGuide.md]: If the tasks are saved after every mutable command, then it does not really matter that the data erased, as the next command will save a new file.  Or are you mentioning about erasing the data before executing the program?   

You also missed out on the feature to save in a different file/directory.

File: docs/UserGuide.md [hunk 22]
```diff
@@ -1,186 +1,239 @@
...
 
 ## 3. FAQ
```
Comment (hwkchia) [inline docs/UserGuide.md]: Seems like your FAQ still assumes GUI event-based input like clicking?

**Review by hwkchia — COMMENTED** [review]
This User Guide seems to have been hastily hacked out.  Typically you have till the end of the week to polish up your deliverable for review, so you have time to make changes.

Moreover, bear in mind that there is a lot of focus on the command line parser in this project, so try to make the command as flexible, and as natural, as you can.

You can refer to more guidance here. 
https://drive.google.com/open?id=1kAth-pTOzwCJFVjS6KU8O50bCvijDqpb6GsOyJTD0Ds

**Comment by hwkchia** [comment]
User Guide has been reviewed.  DevGuide will be reviewed soon, so do not close the PR yet.

File: docs/UserGuide (original).md (hunk 1 above)
Comment (dylansyj) [reply docs/UserGuide (original).md]: Ok, we will remove it. Thank you

File: docs/UserGuide.md (hunk 2 above)
Comment (dylansyj) [reply docs/UserGuide.md]: sorry for the mistake, we will amend it. Thank you!

**Pull request closed without merging**